test/e2e/End2EndSpec.scala

```diff
@@ -43,4 +47,50 @@ class End2EndSpec(arguments: Arguments) extends Specification with GuiceFakeAppl
     customArgumentsMap.groupBy(_(0).substring(2)).view.mapValues(_(0).last).toMap
   }
 
+  private def ensureTestDataset(): Unit = {
+    val testDatasetPath = "test/dataset/test-dataset.zip"
+    val dataDirectory = new File("binaryData/Organization_X")
+    if (!dataDirectory.exists()) {
+      throw new Exception("Organization_X directory does not exist.")
+    }
+    val testDatasetZip = new File(testDatasetPath)
+    if (!testDatasetZip.exists()) {
+      throw new Exception("Test dataset zip file does not exist.")
+    }
+    // Stop if the test dataset is already unzipped
+    if (dataDirectory.listFiles().exists(_.getName == "test-dataset")) {
+      return
```

Comment:
please avoid `return` in scala, always use full if/else instead. compare https://tpolecat.github.io/2014/05/09/return.html